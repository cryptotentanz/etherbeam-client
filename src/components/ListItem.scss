.my-listitem {
  display: inline-flex;
}
