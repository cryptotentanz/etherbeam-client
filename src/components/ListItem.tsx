import React, { FC, ReactNode } from 'react'
import './ListItem.scss'

interface ListItemProps {
  className?: string
  children?: ReactNode
}

const ListItem: FC<ListItemProps> = ({ className, children }: ListItemProps) => {
  return <li className={['my-listitem', className].join(' ')}>{children}</li>
}

export default ListItem
