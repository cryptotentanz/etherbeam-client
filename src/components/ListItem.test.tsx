import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import ListItem from './ListItem'

describe('<ListItem />', () => {
  const content = 'Content'
  const className = 'my-class'

  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<ListItem>{content}</ListItem>)
    })

    it('should render', () => {
      const li = component.find('li')
      expect(li.prop('children')).toBe(content)
      expect(li.hasClass('my-listitem')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Class', () => {
    beforeAll(() => {
      component = shallow(<ListItem className={className}>{content}</ListItem>)
    })

    it('should render', () => {
      const li = component.find('li')
      expect(li.prop('children')).toBe(content)
      expect(li.hasClass('my-listitem')).toBeTruthy()
      expect(li.hasClass('my-class')).toBeTruthy()
    })
  })
})
