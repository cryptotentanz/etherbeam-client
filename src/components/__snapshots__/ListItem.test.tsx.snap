// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Default should match snapshot 1`] = `
<li
  class="my-listitem "
>
  Content
</li>
`;
